django/middleware/locale.py

```diff
@@ -35,7 +35,12 @@ def process_response(self, request, response):
         urlconf = getattr(request, 'urlconf', settings.ROOT_URLCONF)
         i18n_patterns_used, prefixed_default_language = is_language_prefix_patterns_used(urlconf)
 
-        if response.status_code == 404 and not language_from_path and i18n_patterns_used:
+        if (response.status_code == 404 and
```

Comment:
I thought the if style of the old PR was a bit better (with more equal line lengths).